internal/pkg/template/templates/task/cf.yml

```diff
@@ -122,8 +122,33 @@ Resources:
                       "secretsmanager:ResourceTag/copilot-environment": !Ref Env
                   Resource:
                     - !Sub arn:${AWS::Partition}:secretsmanager:${AWS::Region}:${AWS::AccountId}:secret:*
-          - !Ref AWS::NoValue
-      {{- end}}
+          - PolicyName: 'PullSecrets'
+            PolicyDocument:
+              Version: '2012-10-17'
+              Statement:
+              {{- if isSSMParamFlag .Secrets}}
```

Comment:
An alternative would be to have `SSM` and `SecretsManager` as subfileds of `Secrets` (so instead of a `map`, `Secrets` is probably a `struct` now), then here it could be:
```
{{- if .Secrets.SSM}}
   - Effect: 'Allow'
     Action:
         - ssm:GetParameters
      Resource: {{range $name, $valueFrom := .Secrets}} 
       {{- if isARN $valueFrom }}
        - {{$valueFrom}}
        {{- else }}
         - !Sub arn:${AWS::Partition}:ssm:${AWS::Region}:${AWS::AccountId}:parameter{{$valueFrom}}
        {{- end}} {{- end}}
{{- end}}
```
In `cli/task_run.go`, pass ssm secrets to `Secrets.SSM`, and secretsmanager secrets to `Secrets.SecretsManager`.